www/js/init.js

```diff
@@ -33,8 +33,176 @@
+                setSetting(paramKey, paramVal);
+                params[paramKey] = paramVal;
+            }
+        }
+        matches = regexpUrlParams.exec(window.location.search);
+    }
+    // If we are in the PWA version launched from an extension, send a 'success' message to the extension
+    if (params.referrerExtensionURL && ~window.location.href.indexOf(params.PWAServer)) {
+        var message = '?PWA_launch=success';
+        // DEV: To test failure of the PWA, you could pause on next line and set message to '?PWA_launch=fail'
+        // Note that, as a failsafe, the PWA_launch key is set to 'fail' (in the extension) before each PWA launch
+        // so we need to send a 'success' message each time the PWA is launched
+        var frame = document.createElement('iframe');
+        frame.id = 'kiwixComm';
+        frame.style.display = 'none';
+        document.body.appendChild(frame);
+        frame.src = params.referrerExtensionURL + '/www/index.html' + message;
+        // Now remove redundant frame. We cannot use onload, because it doesn't give time for the script to run.
+        setTimeout(function () {
+            var kiwixComm = document.getElementById('kiwixComm');
+            // The only browser which does not support .remove() is IE11, but it will never run this code
+            if (kiwixComm) kiwixComm.remove();
+        }, 3000);
+    }
+})();
+
+// Since contentInjectionMode can be overriden when returning from remote PWA to extension (for example), we have to prevent an infinite loop
+// with code that warns the user to turn off the App Cache bypass in jQuery mode. Note that to turn OFF the bypass, we have to set the VALUE to true
+params.appCache = params.contentInjectionMode === 'jquery' ? true : params.appCache;
+
+/**
+ * Set the State and UI settings associated with parameters defined above
+ */
+document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
+document.getElementById('disableDragAndDropCheck').checked = params.disableDragAndDrop;
+document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+document.getElementById('titleSearchRangeVal').textContent = params.maxSearchResultsSize;
+document.getElementById('appThemeSelect').value = params.appTheme;
+document.getElementById('useHomeKeyToFocusSearchBarCheck').checked = params.useHomeKeyToFocusSearchBar;
+document.getElementById('openExternalLinksInNewTabsCheck').checked = params.openExternalLinksInNewTabs;
+switchHomeKeyToFocusSearchBar();
+document.getElementById('bypassAppCacheCheck').checked = !params.appCache;
+document.getElementById('appVersion').textContent = 'Kiwix ' + params.appVersion;
+// We check here if we have to warn the user that we switched to ServiceWorkerMode
+// This is only needed if the ServiceWorker mode is available, or we are in an Extension that supports Service Workers
```

Comment:
ACtually, I **think** you can probably move all of this from the line above to the comment below back to the beginning of app.js, but remember to change `setSetting` back to the settingsStore version once you move it back.

Reply:
To be clear, I mean move from line 140 to line 157 back.